### PR TITLE
System refactor

### PR DESCRIPTION
**Pull request by Michieldoesburg**
A package with interfaces has been created. Only interfaces for entity, boundingbox and playerfish yet. Some code and almost all javadoc has been refactored. I cant run maven atm because its missing some file, if someone can run it and check for checkstyle errors that would be great.

### CODE REVIEW

**Comment by Michieldoesburg** [comment]
NB: Many getters and setters now only have javadoc in the interface. Because having that much javadoc in a class (for getters and setters) is nonsensical. If checkstyle gives warnings about this, we should change the checkstyle conditions.

**Comment by SunweiWang** [comment]
nice implementation, in fact, since you have already implemented javadoc in the interface, there is no checkstyle warnings in classes extended from the interfaces. However, there are some checkstyle warnings in interfaces though

File: src/interfaces/BoundingBoxInterface.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * BoundingBoxes (axis-aligned) are used to calculate collisions between fish.
+ * Each fish is represented in the 2-D plane of the screen with a BoundingBox.
+ * 
+ * @author Michiel
+ */
+public interface BoundingBoxInterface {
+
+	  /**
```
Comment (SunweiWang) [inline src/interfaces/BoundingBoxInterface.java]: each of these lines all gives me checkstyle warnings:
"Line contains a tab space"
Does someone wants to fix it or I can do it as well?
< review, layout

**Comment by SunweiWang** [comment]
Btw, I would prefer if we can merge Michiel's commits before I work on my parts, otherwise I can see a lot of merge conflicts coming.
< review, organisation

**Comment by ClintonCao** [comment]
Indeed a nice work with the @inheritDoc. I didn't know that it was possible to do that. I ran maven build on my machine and will fixing the warnings soon. <review, layout

**Comment by Michieldoesburg** [comment]
Youre right sunwei you definitely need the refactor before you can continue with the specific design patterns. 

As an addendum: it has become pretty clear now that the static methods (in PlayerFish and EnemyFish) need to be turned into factories. So its pretty easy to identify methods that might need to become factories (just look for statics).

**Comment by ClintonCao** [comment]
I have fixed the checkstyle warnings for this pull request. After the discussing on the Whatsapp chat and disagreeing on removing the JavaDoc module from checkstyle, a new solution has been proposed by @Michieldoesburg. The solution was to add a new property to JavaDoc module; "allowMissingPropertyJavadoc". Thanks for the solution! This is now added to the checkstyle.xml. 

Also when making interfaces, it is not necessary add the modifiers for the methods (as the concrete classes will be implementing the methods).

The rest seems fine to me and nice job! <layout, documentation, configuration, review

**Pull request merged**